### PR TITLE
Clarify the docs on Node.queue_free

### LINKED ISSUE #77793: Collision exception and node freeing error in Godot 4.0.3
### Godot version

4.0.3

### System information

Win10

### Issue description

I encountered a bug in Godot 4.0.3 when I tried to add a collision exception and free a character body 2D node. The code is as follows:

```gdscript
func _ready():
    character_body_2d.add_collision_exception_with(character_body_2d_2)
    character_body_2d_2.queue_free()
# Called every frame. "delta" is the elapsed time since the previous frame.
func _process(delta):
    character_body_2d.get_collision_exceptions()
```

> The error message is:

```text
ERROR: Condition "!body" is true. Returned: ObjectID()
   at: body_get_object_instance_id (servers/physics_2d/godot_physics_server_2d.cpp:690)
```

> I think this is a bug because the documentation says that `add_collision_exception_with` should prevent collisions between two bodies, and `queue_free` should defer the deletion of a node until it is safe. However, it seems that the node is freed before the collision exception is applied, causing an error when calling `get_collision_exceptions`. Can anyone confirm this issue and provide a workaround? Thank you.

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by RedworkDE**
Fixes #77793

`This method ensures it's safe to delete the node, contrary to [method Object.free].`  can interpreted to mean basically anything so instead list what (deletion related things) you can safely do with the method.

### CODE REVIEW

**Comment by AThousandShips** [comment]
I'd personally say it's safe to just link to `Object.free` to point to the clarification there, but might it be useful to add the "Pre-existing references to the object become invalid, and any attempt to access them will result in a run-time error.", adapted to the queued nature, here as well?

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -654,7 +654,8 @@
 		<method name="queue_free">
 			<return type="void" />
 			<description>
-				Queues a node for deletion at the end of the current frame. When deleted, all of its child nodes will be deleted as well. This method ensures it's safe to delete the node, contrary to [method Object.free]. Use [method Object.is_queued_for_deletion] to check whether a node will be deleted at the end of the frame.
+				Queues a node for deletion at the end of the current frame. When deleted, all of its child nodes will be deleted as well and all references to the node and its children will become invalid, see [method Object.free].
```
Comment (AThousandShips) [inline doc/classes/Node.xml:657]:
```suggestion
				Queues a node for deletion at the end of the current frame. When deleted, all of its child nodes will be deleted as well, and all references to the node and its children will become invalid, see [method Object.free].
```
To help the flow of the sentence

**Review by AThousandShips — APPROVED** [review]
LGTM, helpful wording and think it covers everything

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**